src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEMKernel.hpp

```diff
@@ -0,0 +1,717 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ElasticWaveEquationSEMKernel.hpp
+ */
+
+#ifndef GEOSX_PHYSICSSOLVERS_WAVEPROPAGATION_ELASTICWAVEEQUATIONSEMKERNEL_HPP_
+#define GEOSX_PHYSICSSOLVERS_WAVEPROPAGATION_ELASTICWAVEEQUATIONSEMKERNEL_HPP_
+
+#include "finiteElement/kernelInterface/KernelBase.hpp"
+
+
+namespace geosx
+{
+
+/// Namespace to contain the elastic wave kernels.
+namespace elasticWaveEquationSEMKernels
+{
+
+struct PrecomputeSourceAndReceiverKernel
+{
+  /**
+   * @brief Check if the source point is inside an element or not
+   * @param numFacesPerElem number of face on an element
+   * @param elemCenter array containing the center of the elements
+   * @param faceNormal array containing the normal of all faces
+   * @param faceCenter array containing the center of all faces
+   * @param elemsToFaces map to get the global faces from element index and local face index
+   * @param coords coordinate of the point
+   * @return true if coords is inside the element
+   */
+
+  GEOSX_HOST_DEVICE
+  static bool
+  locateSourceElement( real64 const numFacesPerElem,
+                       real64 const (&elemCenter)[3],
+                       arrayView2d< real64 const > const faceNormal,
+                       arrayView2d< real64 const > const faceCenter,
+                       arraySlice1d< localIndex const > const elemsToFaces,
+                       real64 const (&coords)[3] )
```

Comment:
For my own understanding, why not using and fixing this function [here](https://github.com/GEOSX/GEOSX/blob/36f108a19fec1ffb2429b4e7316c8bcb8fddd999/src/coreComponents/mesh/utilities/ComputationalGeometry.hpp#L368) since it is doing the same thing. Or at least using the same code in `AcousticWaveEquationSEMKernel.hpp` and `ElasticWaveEquationSEMKernel.hpp`?

Reply:
It is because of high order. centroid3D function needs to be changed to take in account the new nodes due to high order but if we change it it broke integrated test because of solver using tetrahedra. So to not have to modify a lot of stuff I changed it locally on our solvers (now that a part of high order is merged, both solvers should be the same)